proto/store/database.proto

```diff
@@ -325,6 +325,10 @@ message IndexMetadata {
   // This could refer to a column or an expression.
   repeated string expressions = 2;
 
+  // The key_lengths are the ordered key lengths of an index.
+  // If the key length is not specified, it's -1.
+  repeated int64 key_length = 9;
```

Comment:
use int32 if we can. easier to handle on the frontend.

Reply:
and i think we can
https://docs.pingcap.com/tidb/v5.0/tidb-configuration-file#max-index-length